lib/fs.js

```diff
@@ -1319,14 +1318,14 @@ fs.watchFile = function(filename) {
     persistent: true
   };
 
-  if (arguments[1] !== null && typeof arguments[1] === 'object') {
-    options = util._extend(options, arguments[1]);
-    listener = arguments[2];
+  if (options !== null && typeof options === 'object') {
+    options = util._extend(defaults, options);
   } else {
-    listener = arguments[1];
+    listener = options;
+    options = defaults;
   }
 
-  if (!listener) {
+  if (typeof listener !== 'function') {
     throw new Error('watchFile requires a listener function');
```

Comment:
Never too late :) you're right though, if you wanted to submit a PR I'd review it.